src/core/DataFile.cpp

```diff
@@ -1684,6 +1685,43 @@ void DataFile::upgrade_noteTypes()
 	}
 }
 
+void DataFile::upgrade_fixCMTDelays()
+{
+	static QMap<QString, QString> nameMap;
+	nameMap["delay_0,01s"] = "delay_0.01s";
+	nameMap["delay_0,1s"] = "delay_0.1s";
+	nameMap["fbdelay_0,01s"] = "fbdelay_0.01s";
+	nameMap["fbdelay_0,1s"] = "fbdelay_0.1s";
+
+	const auto effects = elementsByTagName("effect");
+
+	for (int i = 0; i < effects.size(); ++i)
+	{
+		auto effect = effects.item(i).toElement();
+
+		// We are only interested in LADSPA plugins
+		if (effect.attribute("name") != "ladspaeffect") { continue; }
+
+		// Fetch all attributes (LMMS) beneath the LADSPA effect so that we can check the value of the plugin attribute (XML)
+		auto attributes = effect.elementsByTagName("attribute");
+		for (int j = 0; j < attributes.size(); ++j)
+		{
+			auto attribute = attributes.item(j).toElement();
+
+			if (attribute.attribute("name") == "plugin")
+			{
+				const auto attributeValue = attribute.attribute("value");
+
+				const QMap<QString, QString>::const_iterator it = nameMap.find(attributeValue);
+				if (it != nameMap.end())
```

Comment:
This should work
```suggestion
				auto it = nameMap.constFind(attributeValue);
				if (it != nameMap.constEnd())
```

Reply:
Adjusted with commit 564f50f2e37. I have used `const auto` because with only `auto` it was still possible to increment the iterator. Now the thing is as `const` as it gets. :sweat_smile:

Reply:
Good point! If only... `const auto const it const = const nameMap.constFind(const attributeValue) const;` 😂